JSONObject.java

```diff
@@ -1588,6 +1588,22 @@ public JSONObject put(String key, Map<?, ?> value) throws JSONException {
         return this;
     }
 
+    /**
+     * Put a key/char pair in the JSONObject.
+     *
+     * @param key
+     *            A key string.
+     * @param value
+     *            A char which is the value.
+     * @return this.
+     * @throws JSONException
+     *             If the key is null.
+     */
+    public JSONObject put(String key, char value) throws JSONException {
+	    this.put(key, (Character)value);
+        return this;
+    }
+
     /**
      * Put a key/value pair in the JSONObject. If the value is null, then the
```

Comment:
I'd rather have this be `Character.valueOf(value)` instead of a cast.

Reply:
Either way...

Reply:
The valueOf method should offer caching on most jvm implementations while the cast never will. The valueOf should be preferable for any of the primitive to object boxing conversions.

Reply:
Okay.  Will do.